var/spack/repos/builtin/packages/fenics/package.py

```diff
@@ -43,135 +61,105 @@ class Fenics(CMakePackage):
             values=('Debug', 'Release', 'RelWithDebInfo',
                     'MinSizeRel', 'Developer'))
 
-    # not part of spack list for now
-    # variant('petsc4py',     default=True,  description='Uses PETSc4py')
-    # variant('slepc4py',     default=True,  description='Uses SLEPc4py')
-    # variant('pastix',       default=True,  description='Compile with Pastix')
-
-    patch('petsc-3.7.patch', when='@1.6.1^petsc@3.7:')
-    patch('petsc-version-detection.patch', when='@:1.6.1')
-    patch('hdf5~cxx-detection.patch')
-
-    extends('python')
-
+    # Conflics for PETSC4PY / SLEPC4PY
+    conflicts('+petsc4py', when='~python')
+    conflicts('+petsc4py', when='~petsc')
+    conflicts('+slepc4py', when='~python')
+    conflicts('+slepc4py', when='~slepc')
+
+    # Patches
+    # patch('petsc-3.7.patch', when='petsc@3.7:')
+    patch('petsc-version-detection.patch', when='@1.6.0')
+    patch('hdf5~cxx-detection.patch', when='@:1.6.0')
+
+    patch('header_fix.patch', when='@2019.1.0.post0')
+    patch('petsc_3_11.patch', when='@2018.1.0.post1')
+
+    # enable extension support for fenics package
+    extends('python', when='+python')
+
+    # fenics python package dependencies
+    for ver in dolfin_versions:
+        wver = '@' + ver
+        depends_on('py-fenics-fiat{0}'.format(wver), type=('build', 'run'), when=wver)
+        if(Version(ver) < Version('2018.1.0')):
+            depends_on('py-fenics-instant{0}'.format(wver), type=('build', 'run'), when=wver)
+        else:
+            depends_on('py-fenics-dijitso{0}'.format(wver), type=('build', 'run'), when=wver)
+        depends_on('py-fenics-ufl{0}'.format(wver), type=('build', 'run'), when=wver)
+        if ver in ['2019.1.0', '2017.2.0']:
+            wver = wver + '.post0'
+        depends_on('py-fenics-ffc{0}'.format(wver), type=('build', 'run'), when=wver)
```

Comment:
Are these only needed when `+python`?

Reply:
Checking the [CMakeLists.txt](https://bitbucket.org/fenics-project/dolfin/src/2019.1.0.post0/CMakeLists.txt)  for `dolfin`, the python modules FFC, FIAT, Dijitso (Instant) and UFL are only required for `dolfin` when build with the python interface. 

Can I add the `+python` to the version dependency in `when=`?

Reply:
Yes, it would look like `when='@1.2.3: +python'`